src/binding/c/topo_api.txt

```diff
@@ -0,0 +1,154 @@
+     case.
+*/
+{ -- error_check --
+    MPIR_Topology *cart_ptr = MPIR_Topology_get(comm_ptr);
+    MPIR_ERR_CHKANDJUMP((!cart_ptr || cart_ptr->kind != MPI_CART),
+                        mpi_errno, MPI_ERR_TOPOLOGY, "**notcarttopo");
+    int ndims = cart_ptr->topo.cart.ndims;
+    if (ndims != 0) {
+        MPIR_ERRTEST_ARGNULL(coords, "coords", mpi_errno);
+    }
+    for (int i = 0; i < ndims; i++) {
+        if (!cart_ptr->topo.cart.periodic[i]) {
+            int coord = coords[i];
+            MPIR_ERR_CHKANDJUMP3((coord < 0 || coord >= cart_ptr->topo.cart.dims[i]),
+                                  mpi_errno, MPI_ERR_ARG, "**cartcoordinvalid",
+                                  "**cartcoordinvalid %d %d %d", i, coords[i],
+                                  cart_ptr->topo.cart.dims[i] - 1);
+        }
+    }
+}
+
+MPI_Cart_shift:
+    .desc: Returns the shifted source and destination ranks, given a shift direction and amount
+    .skip: global_cs
+    .extra: ignore_revoked_comm
+/*
+    Notes:
+    The 'direction' argument is in the range '[0,n-1]' for an n-dimensional
+    Cartesian mesh.
+*/
+{ -- error_check -- disp
+}
+
+MPI_Cart_sub:
+    .desc: Partitions a communicator into subgroups which form lower-dimensional cartesian subgrids
+
+MPI_Cartdim_get:
+    .desc: Retrieves Cartesian topology information associated with a communicator
+    .skip: global_cs
+    .extra: ignore_revoked_comm
+
+MPI_Dims_create:
+    .desc: Creates a division of processors in a cartesian grid
+
+MPI_Dist_graph_create:
+    .desc: MPI_DIST_GRAPH_CREATE returns a handle to a new communicator to which the distributed graph topology information is attached
+    .extra: errtest_comm_intra
+
+MPI_Dist_graph_create_adjacent:
+    .desc: returns a handle to a new communicator to which the distributed graph topology information is attached
+    .skip: validate-DEGREE
+    .extra: errtest_comm_intra
+
+MPI_Dist_graph_neighbors:
+    .desc: Provides adjacency information for a distributed graph topology
+
+MPI_Dist_graph_neighbors_count:
+    .desc: Provides adjacency information for a distributed graph topology
+    .extra: ignore_revoked_comm
+
+MPI_Graph_get:
+    .desc: Retrieves graph topology information associated with a communicator
+    .skip: global_cs
+    .extra: ignore_revoked_comm
+
+MPI_Graph_map:
+    .desc: Maps process to graph topology information
+    .skip: global_cs
+    .impl: topo_fns->graphMap
+{ -- error_check --
+    MPIR_ERR_CHKANDJUMP(comm_ptr->local_size < nnodes, mpi_errno, MPI_ERR_ARG, "**graphnnodes");
+}
+
+MPI_Graph_neighbors:
+    .desc: Returns the neighbors of a node associated
+    .skip: global_cs
+
+MPI_Graph_create:
+    .desc: Makes a new communicator to which topology information
+    .docnotes: errtest_comm_intra
+    .impl: topo_fns->graphCreate
+/*
+    Notes:
+    Each process must provide a description of the entire graph, not just the
+    neigbors of the calling process.
```

Comment:
It's not your mistake, but can you change `neigbors` to `neighbors`?

Reply:
Fixed.